Port/pose writer

vitoranello opened the pull request:
Port pose_writer file to ROS2

vitoranello committed:
Initial port pose_writer

vitoranello committed:
:white_check_mark: Port Pose_writer to ROS2

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
+    by entering a name. The saved poses are stored in a YAML file for later use.
+    '''
+    
+    def __init__(self, node_name: str = 'pose_writer'):
```
I don't see a point in having the node name as a parameter if we're not using this class outside of this file.

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
         self.yaml_path = self.config_path + '/' + self.yaml_file
+        self.declare_parameter('~pose_topic', '/amcl_pose')
+        self.pose_topic = self.get_parameter('~pose_topic').get_parameter_value().string_value
+        print(f"Pose topic: {self.pose_topic}")
```
I would just remove this print. If you want to keep it change it to get_logger().info

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
-        self.pose_topic = rospy.get_param('~pose_topic', 'amcl_pose') 
-
-        self.pose_sub = rospy.Subscriber(self.pose_topic, PoseWithCovarianceStamped, self.pose_callback)
+        self.pose_sub = self.create_subscription(PoseWithCovarianceStamped, '/amcl_pose', self.pose_callback, 10)
```
you're using '/amcl_pose' directly here instead of the parameter pose_topic.

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -78,48 +109,55 @@ def save_pose(self):
-                    rospy.logwarn("Invalid input. Please enter 'y' or 'n'.")
+                    self.get_logger().info("Invalid input. Please enter 'y' or 'n'.")
 
     def write_to_yaml(self):
```
Add documentation to this method.

gadorneles replied:
```suggestion
    def write_to_yaml(self) -> None:
```

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
+    def __init__(self, node_name: str = 'pose_writer'):
+        '''
+        @brief Constructor for the PoseWriter node.
+        @param node_name The name of the ROS2 node.
```
Need a colon after param name.

gadorneles replied:
```suggestion
        @param node_name: The name of the ROS2 node.
```

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
     def pose_callback(self, msg):
+        '''
+        @brief Callback function for the pose subscriber.
+        @param msg The PoseWithCovarianceStamped message received from the topic.
```
Need a colon after param name.

gadorneles replied:
```suggestion
        @param msg: The PoseWithCovarianceStamped message received from the topic.
```

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
-        self.pose_sub = rospy.Subscriber(self.pose_topic, PoseWithCovarianceStamped, self.pose_callback)
+        self.pose_sub = self.create_subscription(PoseWithCovarianceStamped, '/amcl_pose', self.pose_callback, 10)
 
     def pose_callback(self, msg):
```
```suggestion
    def pose_callback(self, msg: PoseWithCovarianceStamped) -> None:
```

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
+        self.get_logger().info(f"Received msg: {msg.pose.pose.position.x}, {msg.pose.pose.position.y}, {msg.pose.pose.position.z}")
         self.current_pose = msg.pose.pose
 
     def save_pose(self):
```
```suggestion
    def save_pose(self) -> None:
```

jardeldyonisio commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,105 @@
+        self.poses = {'pose': {'ros__parameters': {'targets': {}}}}
 
-        self.config_path = self.root_path + '/fbot_ws/src/butia_world/config'
+        self.config_path = self.root_path + '/fbot_ws/src/fbot_world/fbot_world_bringup/config'
```
Maybe it's better get the config path using os (import os)

vitoranello committed:
:fire: Remove node_name on the Class PoseWriter
:fire: Remove print
:bug: Change '/amcl_pose' on line 70 to self.pose_topic
:bug: Add a type of msg on pose_callback

jardeldyonisio commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -78,48 +106,58 @@ def save_pose(self):
                     existing_data = yaml.load(yaml_file, Loader=OrderedLoader) or OrderedDict()
                 except yaml.YAMLError as e:
-                    rospy.logerr(f"Error reading {self.yaml_file}: {e}")
+                    self.get_logger().info(f"Error reading {self.yaml_file}: {e}")
```
Log error

jardeldyonisio commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -78,48 +106,58 @@ def save_pose(self):
                     break
                 else:
-                    rospy.logwarn("Invalid input. Please enter 'y' or 'n'.")
+                    self.get_logger().info("Invalid input. Please enter 'y' or 'n'.")
```
Log warning

jardeldyonisio commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,102 @@
 
             if not hasattr(self, 'current_pose'):
-                rospy.logwarn("No pose received from topic yet.")
+                self.get_logger().info("No pose received from topic yet.")
```
Log warning

jardeldyonisio commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -1,74 +1,102 @@
 
             if not pose_name:
-                rospy.logwarn("No name provided, skipping pose.")
+                self.get_logger().info("No name provided, skipping pose.")
```
Log warning

jardeldyonisio commented on fbot_world_bringup/scripts/teste.py:
Remove this test file

vitoranello committed:
Change log's ingo to log's warming

vitoranello committed:
Change  config file path to use os.path.join

gadorneles commented on fbot_world_bringup/nodes/pose_writer.py:
```diff
@@ -78,48 +106,58 @@ def save_pose(self):
 
         if os.path.exists(self.yaml_path):
-            rospy.loginfo(f"{self.yaml_file} already exists. The new poses will be appended to the existing data.")
+            self.get_logger().error(f"{self.yaml_file} already exists. The new poses will be appended to the existing data.")
```
This should not be an error, just an info. It's a common case to want to just append new poses instead of creating a new file.

vitoranello committed:
Change get warning to get info in line 129

jardeldyonisio reviewed (approved):
Great work!

gadorneles reviewed (approved):
Great Work!! :tada:
